docs/apidiffs/current_vs_latest/opentelemetry-sdk-trace.txt

```diff
@@ -1,2 +1,8 @@
 Comparing source compatibility of opentelemetry-sdk-trace-1.44.0-SNAPSHOT.jar against opentelemetry-sdk-trace-1.43.0.jar
-No changes.
\ No newline at end of file
++++  NEW INTERFACE: PUBLIC(+) ABSTRACT(+) io.opentelemetry.sdk.trace.data.ExceptionEventData  (not serializable)
+	+++  CLASS FILE FORMAT VERSION: 52.0 <- n.a.
+	+++  NEW INTERFACE: io.opentelemetry.sdk.trace.data.EventData
+	+++  NEW SUPERCLASS: java.lang.Object
+	+++  NEW METHOD: PUBLIC(+) STATIC(+) io.opentelemetry.sdk.trace.data.ExceptionEventData create(io.opentelemetry.sdk.trace.SpanLimits, long, java.lang.Throwable, io.opentelemetry.api.common.Attributes)
+	+++  NEW METHOD: PUBLIC(+) ABSTRACT(+) io.opentelemetry.api.common.Attributes getAdditionalAttributes()
```

Comment:
Upon closer inspection, I find this method strange. The resulting `ExceptionEventData` interface has the following methods:
- `getAttributes()`: default implementation returns `getAdditionalAttributes()` provided by user + `exception.*` attributes following semconv.
- `getAdditionalAttributes()`: default implementation returns only additional attributes provided by the user.

Seems odd to have a special accessor for `getAdditionalAttributes()`. I assume that consumers of this want to do something like: `if(event instanceOf ExceptionEventData) serialize(((ExceptionEventData) event).getException())`

If they want o split out the exception attributes from additional attributes, can do so with `getAttributes().toBuilder().removeIf(attributeKey -> attributeKey.getKey().startsWith("exception")).build()`.

Do we really need the special `getAdditionalAttributes()` accessor?

Reply:
hmm. good eye on double checking the original API for this. Maybe we're not quite ready to stabilize, if we're unsure of the final API? I agree the creational patterns for this are currently strange and not user-friendly. 🤔

Reply:
I think we can still stabilize, but cut out the `getAdditionalAttributes()` method. The point I make [here](https://github.com/open-telemetry/opentelemetry-java/pull/6795#discussion_r1809033321) about `SpanLimits` as a constructor argument is an implementation detail. We can remove it as a parameter with the only difference being that we compute the `exception.*` attributes on the calling thread instead of on a background thread, which seems pretty reasonable / expected.

Reply:
If you agree, I can either push some commits to this PR, or provide some more complete guidance to @vasantteja on how to proceed.

Reply:
I'm totally ok with your suggestion

Reply:
@jack-berg I understood that we should be removing `getAdditionalAttributes()` from the implementation. Apart from this should I remove anything else?

Reply:
You've got it. 

The `exception.*` attributes will need to be computed in `SdkSpan`, rather than in `ImmutableExceptionEventData`.

Reply:
@jack-berg  I finally wrapped my head around this and have an implementation. I cut out the attribute computation in the `ImmutableExceptionEventData` and I moved them to `SdkSpan`. I added a little bit of logic to shrink the exception messages if maxLength attribute is turned on in `SpanLimits`. Let me know if this aligns with your thinking.